src/main/java/com/arpnetworking/clusteraggregator/aggregation/AggMessageExtractor.java

```diff
@@ -52,6 +53,18 @@ public String entityId(final Object message) {
                     .append(metricData.getMetric())
                     .append("||")
                     .append(metricData.getPeriod());
+            for (final Messages.DimensionEntry dimensionEntry : metricData.getDimensionsList()) {
+                final String k = dimensionEntry.getKey();
+                if (!(k.equals(AggClientConnection.CLUSTER_KEY)
```

Comment:
The keys need to be processed in a fixed+repeatable order (e.g. sorted). If you are going to sort them you could probably remove the special keys at the same time (if there is such an extensible sorting algorithm available).